pkg/routing/node.go

```diff
@@ -69,6 +71,17 @@ func NewLocalNode(conf *config.Config) (*LocalNodeImpl, error) {
 		l.node.Ip = conf.RTC.NodeIP
 		l.node.Region = conf.Region
 	}
+
+	// set up stats history to be able to measure different rate windows
+	var maxInterval time.Duration
+	for _, rateInterval := range conf.NodeStats.StatsRateMeasurementIntervals {
+		if rateInterval > maxInterval {
+			maxInterval = rateInterval
+		}
+	}
+	l.statsHistory = make([]*livekit.NodeStats, (maxInterval+conf.NodeStats.StatsUpdateInterval-1)/conf.NodeStats.StatsUpdateInterval)
```

Comment:
should the update interval be automatically chosen as the greatest common divisor of the stats intervals to prevent misconfiguration?

Reply:
Thought about that. But, the default use is one rate interval which is 10s. So, the GCD would be 10s. But, wanted to update the stats much quicker than that (every 2 seconds by default). Effectively creating a sliding window updated every 2s with 8 second overlap. Gut feel: there is a nicer config to achieve that, but I just could not hit the right/cleaner config to achieve that. Will think more about it. Please let me know if you have ideas.

(made the proto and config more flexible as I was making relevant changes even though a single rate is what will most probably get used).

Reply:
Will merge this, but will think more about the config and make a subsequent PR when something better comes.